util/preprocess.py

```diff
@@ -41,10 +41,9 @@ def preprocess(csv_files, batch_size, numcep, numcontext, alphabet, hdf5_cache_p
             transcript = file.root.transcript[:]
             transcript_len = file.root.transcript_len[:]
 
-            # features are stored flattened, so reshape into
-            # [n_steps, (n_input + 2*n_context*n_input)]
+            # features are stored flattened, so reshape into [n_steps, numcep]
             for i in range(len(features)):
-                features[i] = np.reshape(features[i], [features_len[i], -1])
+                features[i].shape = [features_len[i]+2*numcontext, numcep]
```

Comment:
Setting `.shape` instead of using `np.reshape` guarantees it won't copy the data, it'll instead fail if the reshape can't be done without copying.